src/frontend/src/app/problem/code-editor/code-editor.component.ts

```diff
@@ -24,7 +29,48 @@ export class CodeEditorComponent {
     },
   };
 
-  onCodeChanged(value: any) {
-    console.log('CODE', value);
+  solution = "";
+  constructor(private submitSolutionService: SolutionSubmitService, public dialog: MatDialog) {
+  }
+
+  ngOnInit() {
+    this.solution = this.codeModel.value;
+    this.codeModel.value = prettier.format(this.codeModel.value, {
+      parser: "babel",
+      plugins: [tsParser],
+    });
+  }
+
+  onCodeChanged(value: string) {
+    this.solution = value;
+  }
+
+  submitCode() {
+    this.submitSolutionService.submitSolution(this.id, this.solution, this.codeModel.language).subscribe(
```

Comment:
```suggestion
    this.submitSolutionService.submitSolution(this.id, this.solution, this.codeModel.fileType ).subscribe(
```

If above change is made